### DISCUSSION TITLE
compatibility with earlier versions of Clang?

### DISCUSSION THREAD

**Discussion by glevner** [microsoft/STL#3058] [Q&A] [answered]
Is it possible to configure/build STL for compatibility with a version of Clang earlier than 14? (Or for an earlier version of MSVC, for that matter?) Our project is tied to a specific version, and I would really like to avoid having two different builds of LLVM hanging around in the same project.

**Comment by fsb4000** [comment, marked as answer]
No, it's a hard error without configurations.
https://github.com/microsoft/STL/blob/05e5e6838e87f871666c52e5af004cf0294affba/stl/inc/yvals_core.h#L664-L666

It probably won't work even if you comment out this line, because when we update the compiler, we remove workarounds if something has been fixed in the new version of the compilers(MSVC, EDG, Clang).

Clang 14 and MSVC 19.33 update: https://github.com/microsoft/STL/pull/2861 (you can read "Workaround removals")

**Reply by glevner** [reply]
So... if I need STL for Clang 11 or VS 2019, for example, I need to checkout an older commit of STL, for when that was the latest version?

**Reply by fsb4000** [reply]
Yes.
This issue can help you for finding the right commit: https://github.com/microsoft/STL/issues/386

**Reply by glevner** [reply]
OK, thank you.
